server/src/main/java/org/elasticsearch/script/GeoPointFieldScript.java

```diff
@@ -80,8 +80,6 @@ public interface LeafFactory {
         GeoPointFieldScript newInstance(LeafReaderContext ctx);
     }
 
-    private final GeoPoint scratch = new GeoPoint();
-
```

Comment:
I think we should not be creating a GeoPoint per document here. I wonder if this case invalidates the approach of creating the objects at the end of the stack.